### ISSUE TITLE
Scrollbar flickering when using available region area

### ISSUE THREAD

**Issue by fiserj** [ocornut/imgui#4539]
**Version/Branch of Dear ImGui:**
Version: v1.85 WIP (321b84f)
Branch: master

**Back-end/Renderer/Compiler/OS**
Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Compiler: MSVC v142 (VS 2019)
Operating System: Windows 10

**My Issue/Question:**
When some content is placed or sized based on the available region (height in my case), it's possible that it will at some point exceed the visible area. In that case a scrollbar may be added. However, in next frame, since the available area changed (by scrollbar taking a portion of it), it's possible that the new layout will result in removing the need of the scrollbar. If so, then the situation cyclically repeats.

I understand the issue and how it happens, but am wondering if there's a good fix. Obviously showing the scrollbar always is the easiest solution, but that's more of a workaround. I experimented with detecting the cyclic pattern in returned available region size, but it's still a bit handwavy and also doesn't feel like a proper fix. One can also probably evaluate whether this would happen using the style's parameters.

**Screenshots/Video**
Note that the flickering happens on per-frame basis, but might appear slower in the captured GIF.
![repro](https://user-images.githubusercontent.com/602372/133250311-d61ba149-d605-41c3-be5b-c7d9167d591b.gif)

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```cpp
ImGui::SetNextWindowPos ({ 100, 100 }, ImGuiCond_Once);
ImGui::SetNextWindowSize({ 440, 200 }, ImGuiCond_Once);
if (ImGui::Begin("Flicker Issue", nullptr))
{
    static bool always_on = false;
    ImGui::Checkbox("Scrollbar Always On", &always_on);

    ImGui::BeginChild("Container", {}, false, ImGuiWindowFlags_HorizontalScrollbar | (always_on ? ImGuiWindowFlags_AlwaysHorizontalScrollbar : 0));
    {
        const float size = ImGui::GetContentRegionAvail().y;

        for (int i = 0; i < 3; i++)
        {
            if (i)
            {
                ImGui::SameLine();
            }
                    
            char label[4];
            sprintf(label, "%i", i);
            ImGui::Button(label, { size, size });
        }
    }
    ImGui::EndChild();
}
ImGui::End();
```

**Comment by ocornut** [comment]
> When some content is placed or sized based on the available region (height in my case),

The height of your item based on available width, and available width will be altered if contents is larger than window available height. There's a feedback loop and I am not sure a solution can exist other than locking the visibility of scrollbar (to either off or on), exactly as you did.

(Or potentially if our scrollbars didn't affect available content space by e.g. being transparent and overlayed OVER contents, as some other toolkits are doing, but this only works if you make scrollbar disappear when unused which is not great for our population of users).

**Comment by fiserj** [comment]
Right, I wasn't sure if there wasn't something obvious I missed. I was able to detect this on the client side by keeping the last 4 values of `ImGui::GetContentRegionAvail().y` and checking for the flickering pattern, but it doesn't really seem worth it (and "hackish" as well).

I like the idea of disappearing scrollbar. It could perhaps be exposed via another flag. I'll take a look at it, if it keeps annoying me. For now, I'm just sticking to having the scrollbar always on.

Closing the issue, thanks for the response (and the library!).
